CMakeLists.txt

```
@@ -12,7 +12,7 @@ set(CMAKE_AUTOUIC ON)
 set(CMAKE_AUTOMOC ON)
 set(CMAKE_AUTORCC ON)
 
-set(CMAKE_CXX_STANDARD 14)
+set(CMAKE_CXX_STANDARD 17)
```

```suggestion
set(CMAKE_CXX_STANDARD 14)
```

Let's not update it unless absolutely necessary.

---

Why not use C++17?

I do support using new language features and take full advantage of it. But there must be a solid reason for upgrade like let's say `std::filesystem` (in C++17), or if something is very hard to implement in older version of c++ like iterator over compile time list. Or most important factor could be compile time stuff with constexpr, but it has no effect in GUI programming.

I do not think `_v` vs `::value` is enough of a reason for upgrade now.

---

In addition to all these, there isn't much of language feature changes since C++11 to C++17. Most of them were library changes, next major update C++20, will have enough reason to upgrade. modules, co-routines, ranges are big updates.